### DISCUSSION TITLE
O que vocês estão usando para Backend?

### DISCUSSION THREAD

**Discussion by synga** [frontendbr/forum#696] [Discussão]
Apenas para medir a popularidade das ferramentas que temos hoje em dia.

Trabalho na maioria dos meus projetos usando Firebase, acho bem pratico e agora temos também o Stitch da MongoDB que funciona da mesma forma (ainda não usei pra saber como é). Estou agora dando manutenção em um app que tem backend em C#, asp e Entity e, mesmo sabendo quase que zero disso tudo, estou achando ótimo e bem consistente.

Fora isso estou aprendendo MEAN, apesar de que ouvia MUITO mais sobre ele a 1 ano atrás, ainda acho que pode suprir todas minhas necessidades futuras.

Algo que nunca ouço falarem é sobre Java e Ruby (ou nunca mais ouvi falar sobre).

E vocês? Estão aliando alguma tecnologia de backend ao front? Estão deixando isso pro back mesmo e focando apenas no front?

**Comment by antonielcm** [comment]
Cara, minha stack de back-end basicamente é Ruby (+ Rails/Sinatra) e .NET 5/Core (está muito interessante, vale a pena dar uma olhada). Também andei dando uma bisbilhotada em Elixir, que me parece uma ótima opção dependendo do propósito.

**Comment by rsmelo** [comment]
No meu projeto mais recente a stack de backend é node, express, joi e sequelize. Na necessidade de noSQL provavelmente utilizaria o mongoose no lugar do sequelize.

**Comment by kazzkiq** [comment]
No trabalho (e nos projetos pessoais) nós tentamos ao máximo desacoplar o back-end do front.

No trampo, o back-end de basicamente todos os projetos são em Node.js (TypeScript). O banco de dados varia mais de acordo com o projeto, mas geralmente usamos PostgreSQL ou MySQL. Para servir os arquivos estáticos do front-end, geralmente usamos NGINX com uma configuração bem básica mesmo.

Nos projetos pessoais já da pra sair mais do padrão. Eu sempre usei Node.js (com Koa ou Express), mas ultimamente tenho usado a [Crystal](https://github.com/crystal-lang/crystal) que é uma linguagem **incrível** e extremamente rápida, com uma sintaxe que lembra muito o Ruby.

A comunicação do front-end com o back-end é sempre feita via API REST.

**Comment by jmurowaniecki** [comment]
Nos últimos 6 meses: PHP7 (Lumen/Laravel/λ + MariaDB, Redis).
Último ano: Node (Express + MongoDB) e Python (Tornado/Django + MongoDB).

**Comment by annacruz** [comment]
No trabalho o backend é Python/Django, nos meus projetos pessoais em geral é Ruby (puro ou com algum microframework) ou Go

**Comment by diegonogueirasantos** [comment]
Tenho usado bastante .Net pra backend, mas já vejo um forte movimento pra galera puxar pra Node com express. Já até estamos avaliando as etapas e montando uma micro migração pra teste.

**Comment by houstondapaz** [comment]
Tudo depende do projeto mas...
.net framework para legado, migrando para .net core.
Redis para cache;
Mongodb;
Node com express;
Socket.io para websocket;
Docker e docker cloud para gerenciamento;
Azure como nuvem;

Aliás, pq usar Koa no lugar de express?

**Comment by fredmosc** [comment]
No trabalho é Java (Spring Boot) e pessoais RoR e agora estou começando uns
teste com Node/express

Em qui, 31 de ago de 2017 às 10:34, Anna Cruz <notifications@github.com>
escreveu:

> No trabalho o backend é Python/Django, nos meus projetos pessoais em geral
> é Ruby (puro ou com algum microframework) ou Go
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/frontendbr/forum/issues/696#issuecomment-326296604>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJB79Uu8CF74YsehnRyAin1uRP9CpgYuks5sdrZagaJpZM4PIHVF>
> .
>
-- 
Att,

FM

**Comment by lucasvalhos** [comment]
No meu antigo trampo eu definia o back, e estava usando node em tudo. 
A api principal rodava na amazon, e eu tinha tb muitos serviços pequenos de processamento, que tb fiz os apps em node, e integrava a outros serviços da amazon como o SQS, DynamoDB, S3.
Fiz tb robôs para ler o tráfego dos clientes nas redes sociais e processar o fluxo de mensagens.
Usei node tb como robô para o whatsapp, telegram, integrado a telefonia com o asterisk.
Mas aí troquei de trampo e aqui só fico no front-end e o back é em JAVA (eles usam o vRaptor como framework). E o back integra ao ERP Sapiens que utilizam aqui.

**Comment by felipefialho** [comment]
Nossa stack aqui no Cubo está toda em JavaScript, 

- Angular2 (Front-End)
- Node (Back-End)
- Serveless (Servidor)

**Comment by plcosta** [comment]
Ruby on Rails ou Laravel =)

**Comment by kazzkiq** [comment]
@LFeh Que bacana a parte de back estar em serverless! Nunca tinha ouvido falar de uma empresa BR que já tava usando essa tecnologia.

Vocês notaram alguma diferença de performance, comparado à servidores "comuns"? E sobre os custos... realmente diminuem ou na prática dá meio que na mesma?

Conta mais pra nóis que eu fiquei curioso! 😄

**Comment by ghost** [comment]
Cara, ultimamente tô mexendo mais com Python e JS, mas tô sempre disposto em aprender coisas novas.

**Comment by ghost** [comment]
Tive recentemente uma experiência com Elixir e Phoenix no backend.
Elixir é bem bacana se usado pra o problema certo e vindo do JS o aprendizado foi bem rápido. Gostei bastante do workflow com docker e todo o poder que a linguagem rodando na maquina do Erlang nos dá.

Mas atualmente estou em um novo trampo que tem no backend basicamente .Net e tratando-se da arquitetura atual é aceitável para o que fazemos, ainda não existe nenhuma necessidade para incluir outras linguagens no backend.

**Comment by MarcoWorms** [comment]
Gosto muito do Node.js/MongoDB clássico. Da pra alavancar o poder funcional do JS usando ramda e fazer coisas bem legais no server. Se eu fosse sair disso eu iria ver Elixir provavelmente :)

**Comment by WillianSugiyama** [comment]
Aqui na Integravet, a gente anda usando PHP, mas estamos estudando para mudar pra NodeJS

**Comment by felipefialho** [comment]
@kazzkiq Talvez vale a pena abrir uma nova issue para falar de Serveless hein? 😄

**Comment by JeanLLopes** [comment]
Stack focado na plataforma .Net, back em C#, Redis, Entity Framework e Web API.

**Comment by suissa** [comment]
Node.js + MongoDb/Neo4J/Redis/ElasticSearch

**Comment by suissa** [comment]
@LFeh Function as a Service muito me interessa

**Comment by wmsbill** [comment]
Nada sexy aqui
Front: Backbone
Back: Symphony over PHP7/Go
DB: Elastic search/Redis

**Comment by marcelgsantos** [comment]
Para a parte de back-end utilizo PHP. ❤️ 
[Laravel](https://laravel.com/) quando preciso de um framework full-stack e [Lumen](https://lumen.laravel.com/) ou [Slim](https://www.slimframework.com/) quando preciso de um microframework.

**Comment by waldemirflj** [comment]
BD: MySQL, Redis, Elastic search e MongoDB.
Front: Angular 4.
Backend: Lumen, Slim e NodeJS.

**Comment by yanmagale** [comment]
Nos últimos 6 mêses, Node JS.

Ja fiz projetos usando apenas o express e recentemente fiz um projeto com o [AdonisJS](http://adonisjs.com/). Como eu trabalhei bastante com PHP (na última vez utilizando o Laravel), o Adonis possui uma estrutura muito similar, e tão mágico quanto O laravel rsrsrs.

Na parte de BD: Mongo, Redis, MySQL e Neo4J (algumas vezes)

**Comment by felipefialho** [comment]
@kazzkiq @suissa [Introdução ao Serveless](https://medium.com/cubo-digital/introdu%C3%A7%C3%A3o-ao-serverless-38a3b4eb082d)

**Comment by jonathannto** [comment]
No trabalho estamos usamos Java (Spring Boot), como sou novato vou iniciar um projeto em casa com as mesmas skills para treinar mas mais para frente quero usar node.js em casa.
